lib/ruby_lsp/requests/hover.rb

```diff
@@ -61,34 +67,52 @@ def merge_response!(other)
         self
       end
 
-      sig { params(node: SyntaxTree::Command).void }
-      def on_command(node)
-        message = node.message
-        @response = generate_rails_document_link_hover(message.value, message)
-      end
-
       sig { params(node: SyntaxTree::ConstPathRef).void }
       def on_const_path_ref(node)
-        @response = generate_rails_document_link_hover(full_constant_name(node), node)
+        return if DependencyDetector::HAS_TYPECHECKER
+
+        name = full_constant_name(node)
+        find_in_index(name, node)
       end
 
-      sig { params(node: SyntaxTree::CallNode).void }
-      def on_call(node)
-        message = node.message
-        return if message.is_a?(Symbol)
+      sig { params(node: SyntaxTree::Const).void }
+      def on_const(node)
+        return if DependencyDetector::HAS_TYPECHECKER
 
-        @response = generate_rails_document_link_hover(message.value, message)
+        find_in_index(node.value, node)
       end
 
       private
 
-      sig { params(name: String, node: SyntaxTree::Node).returns(T.nilable(Interface::Hover)) }
-      def generate_rails_document_link_hover(name, node)
-        urls = Support::RailsDocumentClient.generate_rails_document_urls(name)
-        return if urls.empty?
+      sig { params(name: String, node: SyntaxTree::Node).void }
+      def find_in_index(name, node)
+        entries = @index.resolve(name, @nesting)
+        return unless entries
+
+        title = +"```ruby\n#{name}\n```"
+        definitions = []
+        content = +""
+        entries.each do |entry|
+          loc = entry.location
+          file_name = Pathname.new(entry.file_path).basename
```

Comment:
I wonder if this should be a method of `Entry` 🤔

Reply:
Good idea!